doc/contributing/unicode.rst

```diff
@@ -0,0 +1,128 @@
+================
+Unicode handling
+================
+This document explains how Unicode and related issues are handled in CKAN.
+For a general introduction to Unicode and Unicode handling in Python 2 please
+read the `Python 2 Unicode HOWTO`_. Since Unicode handling differs greatly
+between Python 2 and Python 3 you might also be interested in the
+`Python 3 Unicode HOWTO`_.
+
+.. _Python 2 Unicode HOWTO: https://docs.python.org/2/howto/unicode.html
+.. _Python 3 Unicode HOWTO: https://docs.python.org/3/howto/unicode.html
+
+.. note::
+
+    This document describes the intended future state of Unicode handling in
+    CKAN. For historic reasons, some existing code does not yet follow the
+    rules described here.
+
+    *New code should always comply with the rules in this document. Exceptions
+    must be documented.*
+
+
+Overall Strategy
+----------------
+CKAN only uses Unicode internally (``unicode`` on Python 2). Conversion to/from
+ASCII strings happens on the boundary to other systems/libaries if necessary.
+
+
+Encoding of Python files
+------------------------
+Files containing Python source code (``*.py``) must be encoded using UTF-8, and
+the encoding must be declared using the following header::
+
+    # encoding: utf-8
+
+This line must be the first or second line in the file. See `PEP 263`_ for
+details.
+
+.. _PEP 263: https://www.python.org/dev/peps/pep-0263/
+
+
+String literals
+---------------
+String literals are string values given directly in the source code (as opposed
+to strings variables read from a file, received via argument, etc.). In
+Python 2, string literals by default have type ``str``. This has the risk of
+accidentially introducing ASCII strings and makes a future support of Python 3
+difficult (since string literals are interpreted as Unicode in Python 3).
+
+Therefore, the ``unicode_literals`` feature is used to make Python 2 treat
+string literals as Unicode::
```

Comment:
We discussed this again just now. @amercader, @TkTech and I are against using `unicode_literals` and we would prefer to change *all* our literal strings to explicit unicode.

We know that's a big change, and it would need to be done in a number of smaller PRs. Are you willing to help with that?